app/controllers/concerns/defence_request_authorization.rb

```diff
@@ -1,19 +1,24 @@
 module DefenceRequestAuthorization
 
+  private
+
   def find_defence_request
+    defence_request_id = params[:defence_request_id] || params[:id]
```

Comment:
I'm not sure about this, seems it could have unpredictable behaviour? `:id` is pretty generic. Maybe the individual controllers should decide where the `id` comes from?